### PR TITLE
 Add text metrics tests

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/Common/src/SafeNativeMethods.cs [hunk 1]
```diff
@@ -791,11 +791,22 @@ internal static class SafeNativeMethods {
...
         
         public static extern int GetThemeString(HandleRef hTheme, int iPartId, int iStateId, int iPropId, StringBuilder pszBuff, int cchMaxBuffChars);
+
         [DllImport(ExternDll.Uxtheme, CharSet=CharSet.Auto)]
-        
         public static extern int GetThemeDocumentationProperty([MarshalAs(UnmanagedType.LPWStr)] string pszThemeName, [MarshalAs(UnmanagedType.LPWStr)] string pszPropertyName, StringBuilder pszValueBuff, int cchMaxValChars);
-        [DllImport(ExternDll.Uxtheme, CharSet=CharSet.Auto)]
-        
+
+        public static class VisualStyleDocProperty
```
Comment (zsd4yr) [inline src/Common/src/SafeNativeMethods.cs]: can this be internal or protected instead?

File: src/Common/src/SafeNativeMethods.cs [hunk 2]
```diff
@@ -806,13 +817,19 @@ internal static class SafeNativeMethods {
...
         [DllImport(ExternDll.Uxtheme, CharSet=CharSet.Auto)]
-        
         public static extern bool GetThemeSysBool(HandleRef hTheme, int iBoolId);
-        [DllImport(ExternDll.Uxtheme, CharSet=CharSet.Auto)]
-        
+
+        [DllImport(ExternDll.Uxtheme, CharSet=CharSet.Auto)]        
         public static extern int GetThemeSysInt(HandleRef hTheme, int iIntId, ref int piValue);
 
+        public static class VisualStyleSystemProperty
```
Comment (zsd4yr) [inline src/Common/src/SafeNativeMethods.cs]: internal or protected instead?

File: src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/TextMetrics.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage("Microsoft.MSInternal", "CA905:SystemAndMicrosoftNamespacesRequireApproval", Scope = "namespace", Target = "System.Windows.Forms.VisualStyles")]
+
+namespace System.Windows.Forms.VisualStyles
+{
+    public struct TextMetrics
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/TextMetrics.cs]: I am not understanding why you made this / where this is all coming from?

Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/TextMetrics.cs]: I am not understanding why you made this / where this is all coming from?

File: src/Common/src/SafeNativeMethods.cs (hunk 1 above)
Comment (hughbe) [reply src/Common/src/SafeNativeMethods.cs]: This is a class inside an `internal` class - I think we can make this `public` safely. IIRC the guideline is that nested classes inside internal classes should be `public` - I remember @ViktorHofer saying something like this (in the context of Interop code, which is what this is!)

File: src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/TextMetrics.cs (hunk 3 above)
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/TextMetrics.cs]: This file used to here: https://github.com/dotnet/winforms/blob/master/src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleTypesAndProperties.cs

What I've done is rename that file to actually match the class name and move the internal classes to a more appropriate location.

The struct `TextMetrics` has existed before this PR and still exists after. There is NO change to any public APIs in this PR

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/TextMetrics.cs]: (I do understand that adding APIs requires review so I'd never make a change deliberately to any public APIs - remove/add/change - without mentioning this explicitly)

File: src/Common/src/SafeNativeMethods.cs (hunk 1 above)
Comment (ViktorHofer) [reply src/Common/src/SafeNativeMethods.cs]: Correct. See this for more info: https://stackoverflow.com/questions/9302236/why-use-a-public-method-in-an-internal-class?answertab=votes#tab-top

File: src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/TextMetrics.cs (hunk 3 above)
Comment (zsd4yr) [reply src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/TextMetrics.cs]: haha you're good. I honestly just didn't know where TextMetrics was coming from. 

I am in-general for moving classes outside of scripts if there is already a class in that script

**Pull request merged**